File: app/src/main/java/com/example/e_commercemobapp/admin/AdminProductAdapter.java
package com.example.e_commercemobapp.admin;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.e_commercemobapp.R;

import java.util.List;

public class AdminProductAdapter extends RecyclerView.Adapter<AdminProductAdapter.ProductViewHolder> {

    private final Context context;
    private final List<AdminProduct> productList;

    public AdminProductAdapter(Context context, List<AdminProduct> productList) {
        this.context = context;
        this.productList = productList;
    }

    @NonNull
    @Override
    public ProductViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.admin_item_product, parent, false);
        return new ProductViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ProductViewHolder holder, int position) {
        AdminProduct p = productList.get(position);

        holder.name.setText(p.getName());
        holder.category.setText("Category: " + p.getCategoryName());
        holder.price.setText("$" + p.getPrice());

        Glide.with(holder.itemView.getContext())
                .load(p.getImageUrl())
                .placeholder(R.drawable.ic_launcher_background)
                .into(holder.image);

        // 🔥 CLICK → EDIT PRODUCT
        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(context, EditProductActivity.class);
            intent.putExtra("productId", p.getId());
            context.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return productList.size();
    }

    // -------------------------
    // ViewHolder
    // -------------------------
    static class ProductViewHolder extends RecyclerView.ViewHolder {

        ImageView image;
        TextView name, category, price;

        ProductViewHolder(@NonNull View itemView) {
            super(itemView);
            image = itemView.findViewById(R.id.productImage);
            name = itemView.findViewById(R.id.productName);
            category = itemView.findViewById(R.id.productCategory);
            price = itemView.findViewById(R.id.productPrice);
        }
    }
}

File: app/src/main/java/com/example/e_commercemobapp/admin/AdminProductAdapter.kt
package com.example.e_commercemobapp.admin

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.e_commercemobapp.R
import com.example.e_commercemobapp.model.Product

class AdminProductAdapter(
    private val context: Context,
    private val products: MutableList<Product>,
    private val onClick: (Product) -> Unit
) : RecyclerView.Adapter<AdminProductAdapter.AdminProductViewHolder>() {

    inner class AdminProductViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val name: TextView = view.findViewById(R.id.productNameText)
        val category: TextView = view.findViewById(R.id.productCategoryText)
        val price: TextView = view.findViewById(R.id.productPriceText)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AdminProductViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.admin_item_product, parent, false)
        return AdminProductViewHolder(view)
    }

    override fun onBindViewHolder(holder: AdminProductViewHolder, position: Int) {
        val p = products[position]

        holder.name.text = p.name
        holder.category.text = p.categoryId
        holder.price.text = "${p.price} EGP"

        holder.itemView.setOnClickListener {
            onClick(p)
        }
    }

    override fun getItemCount(): Int = products.size
}
